battleship-game/src/components/Board.vue

```diff
@@ -0,0 +1,66 @@
+<template>
+  <div class="board">
+    <h4>Board game {{rows}} x {{columns}}</h4>
+    <div id="board-container">
+      <table id='my-table'>
+      </table>
+    </div>
+  </div>
+</template>
+
+<script>
+import {EventBus} from '@/services/event-bus';
+
+export default {
+  name: 'board',
+  data() {
+    return {
+      rows: 4,
+      columns: 4,
+    }
+  },
+  mounted() {
+    EventBus.$on('rows', (data) => {
+      this.rows = data;
+    })
+    EventBus.$on('columns', (data) => {
+      this.columns = data;
+      this.createTable();
+    })
+    this.createTable();
+  },
+  methods: {
+      createTable(){
+        let table="<tbody>";
+        for (var i = 0; i < this.rows; i++) {
+          table+="<tr>";
+          for (var j = 0; j < this.columns; j++) {
+            table+="<td>x</td>";
+          }
+          table+="</tr>";
+        }
+        table+="</tbody>";
+        document.getElementById("my-table").innerHTML = table;
```

Comment:
use vuejs directives instead of manually updating the DOM, for example:
```html
<table>
  <tr v-for="row in rows">
    <td v-for="column in columns"></td>
  </tr>
</table>
```